eox_nelp/middleware.py

```diff
@@ -6,7 +6,8 @@
     ExtendedProfileFieldsMiddleware: Set extended_profile_fields in registration form.
 """
 from django.conf import settings
-from django.http import Http404
+from django.http import Http404, parse_cookie
+from django.utils.deprecation import MiddlewareMixin
```

Comment:
why are you using deprecation ?

Reply:
Because openedx is using it?
https://github.com/openedx/edx-platform/blob/open-release/palm.master/openedx/core/djangoapps/lang_pref/middleware.py#L5
https://github.com/openedx/edx-platform/blob/open-release/palm.master/openedx/core/djangoapps/dark_lang/middleware.py#L15

Reply:
IMO that's is not a valid reason, probably that is a old middleware that openedx had to update when they made the django upgrade, however this is a new middleware that you're making from scratch

Reply:
@andrey-canon Actually is somehting confusinf in the nomenclature of django.
https://stackoverflow.com/questions/52913333/why-middleware-mixin-declared-in-django-utils-deprecation-py
Is like a help to create new middleware or to support old styles middleware.
https://stackoverflow.com/questions/52913333/why-middleware-mixin-declared-in-django-utils-deprecation-py

Some base middleware of django are inheriting from it.

Reply:
![image](https://github.com/eduNEXT/eox-nelp/assets/36200299/20c082fa-5788-474e-9051-847b1cc38f2d)


> Some base middleware of django are inheriting from it.

This is the same argument that you exposed in the first comment but with django, my answer is the same those are old middlewares that they made compatible with the new middleware structure 
my question would be do you have any reason to use the structure that django deprecated in the past, why do you think the deprecated version is better than the "new" django way ?

Reply:
No, but I think that Django makes this middleware-mixin with the idea of being compatible with both ways: MIDDLEWARE AND MIDDLEWARE_CLASSES. So I prefer the compatibility of both ways for the middleware. 
![image](https://github.com/eduNEXT/eox-nelp/assets/51926076/ad2f4329-3efa-4881-98ad-a7a2f3bea0d1)

Reply:
MIDDLEWARE_CLASSES was deprecated in Django  1.10 and probably you won't find any use of that in later versions

Reply:
d47f188fb210626fb6b2baac46c1f84b09d6b408